### ISSUE TITLE
Inference from frozen_graph yields all black result despite deeplab/vis.py works as intended (frozen_graph uses the same checkpoint)

### ISSUE THREAD

**Issue by Tix00** [tensorflow/tensorflow#53770]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Stock code
- Both on Windows 10 and Google colab
- TensorFlow 1.x on google colab.
- TensorFlow 2.6.2
- TensorFlow version (use command below):

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

I used the deeplabibrary to train my models. This is the script i used to create the checkpoint
` %tensorflow_version 1.x
!pip install tf_slim
from google.colab import drive
drive.mount('/content/drive')
!python /content/drive/MyDrive/models/research/deeplab/train.py --logtostderr \
   --training_number_of_steps=30000 \
   --train_split="train" \
   --model_variant="xception_65" \
   --atrous_rates=6 \
   --atrous_rates=12 \
   --atrous_rates=18 \
   --output_stride=16 \
   --decoder_output_stride=4 \
  --train_crop_size="513,513" \
   --train_batch_size=1 \
   --dataset="pascal_voc_seg" \
   --tf_initial_checkpoint="/content/drive/MyDrive/models/deeplabv3_pascal_train_aug/model.ckpt" \
   --train_logdir="/content/drive/MyDrive/models/checkpoint_exc_all2" \
   --dataset_dir="/content/drive/MyDrive/models/tfrecord_all" \
   --fine_tune_batch_norm=false \
   --initialize_last_layer=true \
   --last_layers_contain_logits_only=false`

Then, this is the working vis.py:

`%tensorflow_version 1.x
!pip install tf_slim
from google.colab import drive
drive.mount('/content/drive')
!python /content/drive/MyDrive/models/research/deeplab/vis.py --logtostderr \
  --vis_split="val" \
  --model_variant="xception_65" \
  --output_stride=16 \
    --atrous_rates=6 \
   --atrous_rates=12 \
   --atrous_rates=18 \
   --output_stride=16 \
  --decoder_output_stride=4 \
  --vis_crop_size="513,513" \
  --min_resize_value=513 \
  --max_resize_value=513 \
  --dataset="pascal_voc_seg" \
  --checkpoint_dir="/content/drive/MyDrive/models/checkpoint_exc_all2" \
  --vis_logdir="/content/drive/MyDrive/models/Result_img_exc_final" \
  --dataset_dir="/content/drive/MyDrive/models/tfrecord_all" \
  --train_crop_size=513 \
  --max_number_of_iterations=1 --eval_interval_secs=0`

As stated i visualize correctly the segmentated output with vis.py.

I tried to export the checkpoint using this:
`%tensorflow_version 1.x
!pip install tf_slim
from google.colab import drive
drive.mount('/content/drive',force_remount=True)
!python /content/drive/MyDrive/models/research/deeplab/export_model.py --logtostderr \
  --model_variant="xception_65" \
   --atrous_rates=6 \
   --atrous_rates=12 \
   --atrous_rates=18 \
   --output_stride=16 \
   --crop_size=513 \
--crop_size=513\
   --num_classes=21\
   --export_path="/content/drive/MyDrive/models/frozen_graph/prova21.pb"\
   --checkpoint_path="/content/drive/MyDrive/models/checkpoint_exc_all2/model.ckpt-30000" \`
But when loading and using the model for inference i get an all blank(0) result. This is the script i use:
`import os

from matplotlib import gridspec
from matplotlib import pyplot as plt
import numpy as np
from PIL import Image
import time
import cv2
from tqdm import tqdm

import tensorflow as tf

# Needed to show segmentation colormap labels

flags = tf.compat.v1.app.flags

FLAGS = flags.FLAGS

flags.DEFINE_string('model_dir', r"C:\Users\User\PycharmProjects\test_model_seg\models\exc65.pb", 'Where the model is')
flags.DEFINE_string('image_dir', "0000631.jpg", 'Where the image is')
flags.DEFINE_string('save_dir', None, 'Dir for saving results')
flags.DEFINE_string('image_name', None, 'Image name')
class DeepLabModel(object):
    """Class to load deeplab model and run inference."""

    INPUT_TENSOR_NAME = 'ImageTensor:0'
    OUTPUT_TENSOR_NAME = 'SemanticPredictions:0'
    INPUT_SIZE = 513

    def __init__(self, model_dir):
        """Creates and loads pretrained deeplab model."""
        self.graph = tf.Graph()

        graph_def = None
        # Extract frozen graph from tar archive.
        model_filename = FLAGS.model_dir
        with tf.compat.v1.gfile.FastGFile(model_filename, 'rb') as f:
            graph_def = tf.compat.v1.GraphDef()
            graph_def.ParseFromString(open("models\exc65.pb", "rb").read())

        if graph_def is None:
            raise RuntimeError('Cannot find inference graph in tar archive.')

        with self.graph.as_default():
            tf.import_graph_def(graph_def, name='')

        self.sess = tf.compat.v1.Session(graph=self.graph)

    def run(self, image):
        """Runs inference on a single image.

        Args:
            image: A PIL.Image object, raw input image.

        Returns:
            resized_image: RGB image resized from original input image.
            seg_map: Segmentation map of `resized_image`.
        """
        width, height = image.size
        resize_ratio = 1.0 * self.INPUT_SIZE / max(width, height)
        target_size = (int(resize_ratio * width), int(resize_ratio * height))
        resized_image = image.convert('RGB').resize(target_size, Image.ANTIALIAS)
        print('Image resized')
        start_time = time.time()
        batch_seg_map = self.sess.run(
            self.OUTPUT_TENSOR_NAME,
            feed_dict={self.INPUT_TENSOR_NAME: [np.asarray(resized_image)]})
        print('Image processing finished')
        print('Elapsed time : ' + str(time.time() - start_time))
        seg_map = batch_seg_map[0]
        return resized_image, seg_map

model = DeepLabModel(FLAGS.model_dir)
print('Model created successfully')

img = Image.open("0000631.jpg")

or_img, seg = model.run(img)

for x in seg:
    print(x)`
Some side notes that may be usefull:

I used transfer training to segment the skin. My dataset is composed of nearly 32k images/segmented mask. I only have 2 classes in the segmented mask.(skin and background). I don't really know whats wrong with my code, perhaps is the way i import it?

Thanks in advance

**Comment by mohantym** [comment]
Hi @Tix00! Can you please share the model file too.

**Comment by Tix00** [comment]
> Hi @Tix00! Could you please share the model file or update the error trace in above template? Thanks!

Do you need the .pb file or the checkpoint?
And there is no error trace. It simply output and all blank(zeros) result. While it is working with the checkpoint model.

https://drive.google.com/file/d/1-5ad_AcTyfEmUWh5Gu7_JCm-Vu7ZPa8P/view?usp=sharing

**Comment by Tix00** [comment]
> Hi @Tix00! Could you please share the model file or update the error trace in above template? Thanks!

Am i missing something? Do you need more information?

**Comment by mohantym** [comment]
@Tix00 !  Its 1.x code actually which you might be trying to run compatibility mode in 2.6 which might be cause of issue. Can you upgrade your code base to 2.7 version and let us know ? Thanks!

**Comment by Tix00** [comment]
The whole code base(so redoing the preprocessing/training/export_graph) or
just trying to use model .pb file?
 Because otherwise I already tried using the latest version of tensor flow
while trying to make an inference Frome the .pb
Does it gives different results to you?

Il mar 18 gen 2022, 05:10 mohantym ***@***.***> ha scritto:

**Comment by mohantym** [comment]
Can you please share the colab gist with results? Thank you!

**Comment by Tix00** [comment]
Yes of course.
https://gist.github.com/Tix00/156ab9226f951475d9684d3f4e7708c6

Thank you for your help

**Comment by mohantym** [comment]
@Saduf2019 ! Could you please look at this issue ? Attaching [gist](https://colab.sandbox.google.com/gist/mohantym/301704959531d6c6e0cf193928689803/github_53770.ipynb#scrollTo=08ddp6sq1Erk) for reference.

**Comment by sushreebarsa** [comment]
@Tix00 Session is `deprecated`. We see that you're using Session which does not work with  eager execution
 (TF V2). To migrate code that uses sessions to TF2, rewrite the code without it. Please see the [migration guide](https://www.tensorflow.org/guide/migrate#1_replace_v1sessionrun_calls) to migrate from 1.x to 2.x.
tf.compat.v1.GraphDef   is also designed for TensorFlow v1. Please refer [this ](https://www.tensorflow.org/api_docs/python/tf/compat/v1/GraphDef)  .


TF v1.x is currently  not actively supported so please try to upgrade to TF v2.4 or later versions.
For any further queries regarding TF v1.x specific issues please post it in [TF forum](https://discuss.tensorflow.org/) where there is a larger community to get you the right help.
Thanks!

**Comment by Tix00** [comment]
Thanks for your concern. I resolved the issue. 

--save_inference_graph=True \

Adding this flag in the export model script will do the trick.

Have a nice day and thanks for your time
